source/common/network/io_socket_error_impl.cc

```diff
@@ -8,7 +8,34 @@
 namespace Envoy {
 namespace Network {
 
-Api::IoError::IoErrorCode IoSocketError::getErrorCode() const {
+Api::IoError::IoErrorCode IoSocketError::getErrorCode() const { return error_code_; }
+
+std::string IoSocketError::getErrorDetails() const { return errorDetails(errno_); }
+
+IoSocketError* IoSocketError::getIoSocketEagainInstance() {
+  static auto* instance = new IoSocketError(SOCKET_ERROR_AGAIN, Api::IoError::IoErrorCode::Again);
+  return instance;
+}
+
+void IoSocketError::deleteIoError(Api::IoError* err) {
+  ASSERT(err != nullptr);
```

Comment:
Pre-existing: It would be prudent to verify that getIoSocketInvalidAddressInstance() is not being deleted.

`ASSERT(err != getIoSocketInvalidAddressInstance());`

Reply:
Fixed in febdfbc88c7bdf0f2ef1e6baf4156d5e89610bb1